core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java

```diff
@@ -171,6 +176,13 @@ public boolean equals(AsciiString a, AsciiString b) {
         HTTP2_TO_HTTP_HEADER_DISALLOWED_LIST.add(ExtensionHeaderNames.SCHEME.text(), EMPTY_STRING);
         HTTP2_TO_HTTP_HEADER_DISALLOWED_LIST.add(ExtensionHeaderNames.PATH.text(), EMPTY_STRING);
 
+        // https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.3
+        // Request Pseudo-Headers are not allowed for response headers.
+        HTTP2_RESPONSE_HEADERS_DISALLOWED_LIST.add(HttpHeaderNames.AUTHORITY, EMPTY_STRING);
+        HTTP2_RESPONSE_HEADERS_DISALLOWED_LIST.add(HttpHeaderNames.METHOD, EMPTY_STRING);
+        HTTP2_RESPONSE_HEADERS_DISALLOWED_LIST.add(HttpHeaderNames.PATH, EMPTY_STRING);
+        HTTP2_RESPONSE_HEADERS_DISALLOWED_LIST.add(HttpHeaderNames.SCHEME, EMPTY_STRING);
```

Comment:
Instead of doing this, how about checking if there's a header that starts with `:` to be future-proof?

Reply:
Makes sense except for `:status`. Additional response headers could disallow all headers.